packages/nameguard-react/src/components/ReportShield/index.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { type ReactNode } from "react";
+import {
+  CheckResultCode,
+  ConsolidatedNameGuardReport,
+} from "@namehash/nameguard";
+import cc from "classcat";
+
+import { Tooltip } from "../Tooltip/Tooltip";
+import { RatingIcon, RatingIconSize } from "../Report/RatingIcon";
+import { LoadingShieldIcon } from "../icons/LoadingShieldIcon";
+import { UnknownShieldIcon } from "../icons/UnknownShieldIcon";
+import { checkResultCodeTextColor, ratingTextColor } from "../../utils/text";
+
+type ReportShieldProps = {
+  data?: ConsolidatedNameGuardReport;
+  children?: ReactNode;
+  disableHover?: boolean;
+  size?: RatingIconSize;
+  hadLoadingError: boolean;
+} & React.ComponentProps;
+
+export function ReportShield({
+  data,
+  children,
+  disableHover,
+  hasLoadingError,
+  size = RatingIconSize.small,
+  ...props
+}: ReportShieldProps) {
+  if (hasLoadingError) {
+    return <UnknownShieldIcon size={size} {...props} />;
+  }
+
+  if (!data) {
+    return <LoadingShieldIcon size={size} {...props} />;
+  }
+
+  const { title, subtitle, rating, risk_count, highest_risk } = data;
+
+  const textClass = cc(["font-semibold mb-1", ratingTextColor(rating)]);
+
+  if (disableHover) {
+    return <RatingIcon rating={rating} size={size} {...props} />;
+  }
+
+  return (
+    <Tooltip trigger={<RatingIcon rating={rating} size={size} {...props} />}>
```

Comment:
There are two components that replicate this code:

`ReportIcon` and `UnknownReportShield` 

After an analysis I realize it would not be valuable to make these 1: logic complexity would increase a lot and code lines that are reused by both of these are less than 5

I decided to go forward with these separated, then

Let me know if anything 😉

Reply:
@FrancoAguzzi Thanks I appreciate that! Let's please rename `UnknownReportShield` though. Would `UnknownReportIcon` be good? Appreciate your advice :+1: